### DISCUSSION TITLE
indi-allsky service failed

### DISCUSSION THREAD

**Discussion by ashokprksh** [aaronwmorris/indi-allsky#977] [General]
Hi I tried to install the indi allsky in ubuntu 22.04 server. I didn't start the indi-allsky service. attached the log below

Oct  8 12:01:26 allsky-askit [ERROR] MainProcess/MainThread allsky.unhandled_exception() #42: '  File "/home/ashsan/indi-allsky/indi_allsky/camera/indi.py", line 11, in <module>\n    import PyIndi\n'
Oct  8 12:01:26 allsky-askit [ERROR] MainProcess/MainThread allsky.unhandled_exception() #42: '  File "/home/ashsan/indi-allsky/indi_allsky/camera/__init__.py", line 1, in <module>\n    from .indi import IndiClient as indi\n'
Oct  8 12:01:26 allsky-askit [ERROR] MainProcess/MainThread allsky.unhandled_exception() #42: '  File "/home/ashsan/indi-allsky/indi_allsky/capture.py", line 24, in <module>\n    from . import camera as camera_module\n'
Oct  8 12:01:26 allsky-askit [ERROR] MainProcess/MainThread allsky.unhandled_exception() #42: '  File "/home/ashsan/indi-allsky/indi_allsky/allsky.py", line 271, in _startCaptureWorker\n    from .capture import CaptureWorker\n'
Oct  8 12:01:26 allsky-askit [ERROR] MainProcess/MainThread allsky.unhandled_exception() #42: '  File "/home/ashsan/indi-allsky/indi_allsky/allsky.py", line 593, in run\n    self._startCaptureWorker()\n'
Oct  8 12:01:26 allsky-askit [ERROR] MainProcess/MainThread allsky.unhandled_exception() #42: '  File "/home/ashsan/indi-allsky/allsky.py", line 104, in <module>\n    action_func(*args_list, **kwargs_dict)\n'
Oct  8 12:01:26 allsky-askit [ERROR] MainProcess/MainThread allsky.unhandled_exception() #37: Value: No module named 'PyIndi'
Oct  8 12:01:26 allsky-askit [ERROR] MainProcess/MainThread allsky.unhandled_exception() #36: Type: <class 'ModuleNotFoundError'>
Oct  8 12:01:26 allsky-askit [ERROR] MainProcess/MainThread allsky.unhandled_exception() #35: An uncaught exception occurred:
Oct  8 12:01:26 allsky-askit [INFO] MainProcess/MainThread allsky._startup() #240: System uptime: 4744s
Oct  8 12:01:26 allsky-askit [INFO] MainProcess/MainThread allsky._startup() #237: System memory: 15818 MB
Oct  8 12:01:26 allsky-askit [INFO] MainProcess/MainThread allsky._startup() #232: System CPUs: 2
Oct  8 12:01:26 allsky-askit [INFO] MainProcess/MainThread allsky._startup() #230: System Type: Generic PC
Oct  8 12:01:26 allsky-askit [INFO] MainProcess/MainThread allsky._startup() #229: Platform: x86_64
Oct  8 12:01:26 allsky-askit [INFO] MainProcess/MainThread allsky._startup() #228: Python version: 3.11.0rc1
Oct  8 12:01:26 allsky-askit [INFO] MainProcess/MainThread allsky._startup() #226: indi-allsky config level: 20230912.0
Oct  8 12:01:26 allsky-askit [INFO] MainProcess/MainThread allsky._startup() #225: indi-allsky release: 7.5
Oct  8 12:01:26 allsky-askit [WARNING] MainProcess/MainThread allsky._expireOrphanedTasks() #987: Expiring orphaned task 8
Oct  8 11:57:46 allsky-askit [ERROR] MainProcess/MainThread allsky.unhandled_exception() #42: '  File "/home/ashsan/indi-allsky/indi_allsky/camera/indi.py", line 11, in <module>\n    import PyIndi\n'
Oct  8 11:57:46 allsky-askit [ERROR] MainProcess/MainThread allsky.unhandled_exception() #42: '  File "/home/ashsan/indi-allsky/indi_allsky/camera/__init__.py", line 1, in <module>\n    from .indi import IndiClient as indi\n'
Oct  8 11:57:46 allsky-askit [ERROR] MainProcess/MainThread allsky.unhandled_exception() #42: '  File "/home/ashsan/indi-allsky/indi_allsky/capture.py", line 24, in <module>\n    from . import camera as camera_module\n'
Oct  8 11:57:46 allsky-askit [ERROR] MainProcess/MainThread allsky.unhandled_exception() #42: '  File "/home/ashsan/indi-allsky/indi_allsky/allsky.py", line 271, in _startCaptureWorker\n    from .capture import CaptureWorker\n'
Oct  8 11:57:46 allsky-askit [ERROR] MainProcess/MainThread allsky.unhandled_exception() #42: '  File "/home/ashsan/indi-allsky/indi_allsky/allsky.py", line 593, in run\n    self._startCaptureWorker()\n'

**Comment by aaronwmorris** [comment]
If you manually start the program, make sure you activate the python virtualenv before doing so.

```
source virtualenv/indi-allsky/bin/activate
```

**Comment by ashokprksh** [comment]
![image](https://github.com/aaronwmorris/indi-allsky/assets/146148912/c501f28f-26cf-4dc3-b0c3-a69df0bbee12)

Hi Aaron, Thanks for responding, I am getting the prompt while installing to choose the pyindi-client version as shown in the picture, I am getting following error if i choose any of the versions. I choose skip to finish the installation but i couldn't start the allsky service either manually or by the webpage.

Collecting pyindi-client
  Cloning https://github.com/indilib/pyindi-client.git (to revision 674706f) to /tmp/pip-install-wjkz6yd5/pyindi-client_70aa0fed183e4be08cebc67008dab9b4
  Running command git clone --filter=blob:none --quiet https://github.com/indilib/pyindi-client.git /tmp/pip-install-wjkz6yd5/pyindi-client_70aa0fed183e4be08cebc67008dab9b4
  WARNING: Did not find branch or tag '674706f', assuming revision or ref.
  Running command git checkout -q 674706f
  Resolved https://github.com/indilib/pyindi-client.git to commit 674706f
  Preparing metadata (setup.py) ... done
Building wheels for collected packages: pyindi-client
  Building wheel for pyindi-client (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [22 lines of output]
      /home/ashsan/indi-allsky/virtualenv/indi-allsky/lib/python3.11/site-packages/setuptools/dist.py:314: InformationOnly: Normalizing 'v1.9.1' to '1.9.1'
        self.metadata.version = self._normalize_version(
      running bdist_wheel
      running build
      running build_ext
      building '_PyIndi' extension
      swigging indiclientpython.i to indiclientpython_wrap.cpp
      swig -python -v -Wall -c++ -threads -I/usr/include -I/usr/include/libindi -I/usr/local/include/libindi -o indiclientpython_wrap.cpp indiclientpython.i
      Language subdirectory: python
      Search paths:
         ./
         /usr/include/
         /usr/include/libindi/
         /usr/local/include/libindi/
         ./swig_lib/python/
         /usr/share/swig4.0/python/
         ./swig_lib/
         /usr/share/swig4.0/
      Preprocessing...
      Starting language-specific parse...
      /usr/include/libindi/indibase.h:10: Error: Syntax error in input(1).
      error: command '/usr/bin/swig' failed with exit code 1
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for pyindi-client
  Running setup.py clean for pyindi-client
Failed to build pyindi-client
ERROR: Could not build wheels for pyindi-client, which is required to install pyproject.toml-based projects


The script exited abnormally, please try to run again...

**Comment by aaronwmorris** [comment]
It looks like the INDI developers have already fixed this bug.  I am sure an updated package will be released soon.

https://github.com/indilib/indi/pull/1948

You just need to edit `/usr/include/libindi/indibase.h` and remove the space between % and "include".

**Reply by ashokprksh** [reply]
Thanks, I shall try

**Reply by ashokprksh** [reply]
Thanks for the help. It worked
